core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -2339,6 +2357,15 @@ private boolean anyAnyOperands(RexCall call) {
       return false;
     }
 
+    /** Checks whether the expression is null constant. */
+    private boolean isConstantNullExpression(Expression expression) {
+      if (!(expression instanceof ConstantExpression)) {
+        return false;
+      }
+      ConstantExpression ce = (ConstantExpression) expression;
+      return ce.value == null;
+    }
```

Comment:
I think `isConstantNullExpression` is unnecessary, it seems null value is process in
https://github.com/apache/calcite/blob/master/core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java#L1083.
You can check again.

Reply:
@yanlin-Lynn Yes, we can remove this check. Comments addressed, thanks!